server/data_server.c

```diff
@@ -95,6 +96,10 @@ static struct capture_buffer *data_buffer;
  * data capture is enabled, invalid otherwise. */
 static const struct captured_fields *captured_fields;
 static const struct data_capture *data_capture;
+/* PCAP ARM timestamp */
+static struct timespec ts;
+static struct tm tm;
```

Comment:
No, don't do the formatting here.  Just grab the timestamp and do the formatting at the time of generation (`send_capture_info`).  In other words, the only static variable should be `ts`, and it needs a longer and more descriptive name, being a file level variable.  Probably call it `pcap_arm_ts`, please.